src/ios/ffi_for_sanity_tests.rs

```diff
@@ -32,6 +34,47 @@ struct MyStruct {
     my_str: String,
     my_u8: u8,
 }
+#[repr(u8)]
+#[derive(Debug, Clone)]
+pub enum CoreLogLevel {
+    Trace,
+    Debug,
+    Info,
+    Warn,
+    Error,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct CoreLogMessage {
+    level: CoreLogLevel,
+    text: CFStringRef,
+    time: i64,
+}
+
+
+impl From<CoreLogMessageThreadSafe> for CoreLogMessage{
+    fn from(lts: CoreLogMessageThreadSafe) -> Self {
+        let cf_string = CFString::new(&lts.text);
+        let cf_string_ref = cf_string.as_concrete_TypeRef();
+        ::std::mem::forget(cf_string);
+        CoreLogMessage{
+            level: lts.level,
+            text: cf_string_ref,
+            time: lts.time
+        }
+    }
+}
+
+pub struct CoreLogMessageThreadSafe {
```

Comment:
Why exactly is this called thread safe?

Reply:
To be passed through channel, the struct has to implement Send making it thread safe.
I am open to suggestions for renaming.
PS: this struct is a twin of CoreLogMessage (which does not implement Send due to CFStringRef field)

Reply:
Where does it implement Send?

Reply:
> This trait is automatically implemented when the compiler determines it's appropriate.

[trait.Send.html](https://doc.rust-lang.org/std/marker/trait.Send.html)

Specifically, CoreLogMessage is not Send "compatible" due to having CFStringRef field.

Reply:
I see, thanks!